Kickstarter-iOS/Views/Controllers/SheetOverlayContainerViewController.swift

```diff
@@ -0,0 +1,59 @@
+import UIKit
+import Library
+import Prelude
+
+/**
+ SheetOverlayContainerViewController intended to be used as a container for another view controller
+ that you would like to render as a "sheet" or "card" that partially covers the content beneath it.
+ */
+
+final class SheetOverlayContainerViewController: UIViewController {
```

Comment:
I've given this class some more thought and I have to ask if we even need it? Most of the work is done by the animator anyways...so with the same animator we should be able to present `UINavigationController` or just any other `UIVC` class...the animator would just have to ensure rounding corners and style (but I think that's fine coz it's already doing aplenty.

Reply:
I think the benefit of having this class is that it abstracts the configuration of this behavior, which makes it very simple to use with one line of code:

`let sheetOverlayViewController = SheetOverlayContainerViewController(childViewController: navigationController, offset: 50)`

The `SheetOverlay` view controller takes care of holding a reference to the transition animator and conforming to `UIViewControllerTransitioningDelegate`, something that would have to be done by the presenting view controller if we didn't have the `SheetOverlay` class. 

I'd also be wary of putting too much view controller configuration code in the transition animator - I think we should try to limit the behaviour in the animator objects to purely animation-specific functionality for better separation of concerns.